### PR TITLE
adding latus rectum in parabola file in geometry  module

### PR DESCRIPTION
**Pull request by vishakha2108**
#### Brief description of what is fixed or changed
latus rectum - added to parabola.py

#### Release Notes

- geometry
           * added function latus rectum  in parabola

### CODE REVIEW

**Comment by mohitacecode** [comment]
You have to remove the trailing space on line no. 378.
Install `flake8` then you can use `flake8 file_name.py` it will tell you about these type of error within seconds it will save your time.

File: sympy/geometry/parabola.py [hunk 1]
```diff
@@ -376,6 +376,19 @@ def p_parameter(self):
             p = sign(self.focus.args[1] + y)
         return p * self.focal_length
 
+    @property
+    def latus_rectum(self):
+        """latus rectum of parabola- a special chord that passes through focus
```
Comment (smichr) [inline sympy/geometry/parabola.py]: """returns the line passing through the focus that is perpendicular to the axis of symmetry"""?

File: sympy/geometry/parabola.py [hunk 2]
```diff
@@ -376,6 +376,19 @@ def p_parameter(self):
...
+           Returns
+           =======
+           equation of latus rectum
+           >>>p1 = Parabola(Point(0, 0), Line(Point(5, 8), Point(7, 8)))>>>
+           >>> p1.latus_rectum
+          [Line2D(Point2D(0, 0), Point2D(1, 0))]
+        """
+        s = self.directrix.slope
+        return [Line(self.focus,slope = s)]
```
Comment (smichr) [inline sympy/geometry/parabola.py]: I'm not sure that returning a list is better than returning the line itself.

File: sympy/geometry/parabola.py [hunk 3]
```diff
@@ -376,6 +376,19 @@ def p_parameter(self):
             p = sign(self.focus.args[1] + y)
         return p * self.focal_length
 
+    @property
+    def latus_rectum(self):
+        """latus rectum of parabola- a special chord that passes through focus
+           Returns
+           =======
+           equation of latus rectum
+           >>>p1 = Parabola(Point(0, 0), Line(Point(5, 8), Point(7, 8)))>>>
```
Comment (Sc0rpi0n101) [inline sympy/geometry/parabola.py]:
```suggestion
           >>>p1 = Parabola(Point(0, 0), Line(Point(5, 8), Point(7, 8)))
```

File: sympy/geometry/parabola.py (hunk 1 above)
Comment (vishakha2108) [reply sympy/geometry/parabola.py]: yes

File: sympy/geometry/parabola.py [hunk 4]
```diff
@@ -376,6 +376,19 @@ def p_parameter(self):
             p = sign(self.focus.args[1] + y)
         return p * self.focal_length
 
+    @property
+    def latus_rectum(self):
+        """latus rectum of parabola- a special chord that passes through focus
+           Returns
+           =======
+           equation of latus rectum
+           >>> p1 = Parabola(Point(0, 0), Line(Point(5, 8), Point(7, 8))
```
Comment (Sc0rpi0n101) [inline sympy/geometry/parabola.py]:
```suggestion

           Examples
           =======

           >>> p1 = Parabola(Point(0, 0), Line(Point(5, 8), Point(7, 8)))
```

Comment (Sc0rpi0n101) [reply sympy/geometry/parabola.py]: If you use @smichr 's suggestion, I think it'd be better to change the heading to examples and leave a space above and below the heading.

Also, there should be anther closing parenthesis in the definition of p1.

File: sympy/geometry/parabola.py [hunk 5]
```diff
@@ -376,6 +376,19 @@ def p_parameter(self):
...
+           >>> p1.latus_rectum
+           Line2D(Point2D(0, 0), Point2D(1, 0))
+        """
```
Comment (Sc0rpi0n101) [inline sympy/geometry/parabola.py]:
```suggestion

        """
```

Comment (Sc0rpi0n101) [reply sympy/geometry/parabola.py]: You should leave a blank line at the end of a docstring.

**Review by Sc0rpi0n101 — COMMENTED** [review]
I have suggested a few changes to ensure the docstring follows our guidelines. You can read more about the guideline [here](https://docs.sympy.org/dev/documentation-style-guide.html#docstring-guidelines).

**Comment by mohitacecode** [comment]
I noticed that you are doing changes in your `master branch`.I will suggest you should create a new branch for every pr and do the related changes there.

these are the steps you can follow.
```
git fetch upstream
git merge upstream/master
git checkout -b branch_name
```

File: sympy/geometry/parabola.py [hunk 6]
```diff
@@ -376,6 +376,21 @@ def p_parameter(self):
             p = sign(self.focus.args[1] + y)
         return p * self.focal_length
 
+    @property
+    def latus_rectum(self):
+        """Returns the line passing through the focus that is perpendicular to the axis of symmetry
+
+           Examples
+           =======
```
Comment (Smit-create) [inline sympy/geometry/parabola.py]: Should be:
```
Examples
========
```

File: sympy/geometry/parabola.py [hunk 7]
```diff
@@ -376,6 +376,21 @@ def p_parameter(self):
             p = sign(self.focus.args[1] + y)
         return p * self.focal_length
 
+    @property
+    def latus_rectum(self):
+        """Returns the line passing through the focus that is perpendicular to the axis of symmetry
+
```
Comment (Smit-create) [inline sympy/geometry/parabola.py]: Break long lines into small ones. Each line should be of less than 72 characters, using PEP8

**Comment by Smit-create** [comment]
Please add the relevant test cases of the changes you have made.

File: sympy/geometry/parabola.py [hunk 8]
```diff
@@ -376,6 +376,22 @@ def p_parameter(self):
...
+        """Returns the line passing through the focus
+           that is perpendicular to the axis of symmetry
+
+           Examples
+           ========
+
+           >>> p1 = Parabola(Point(0, 0), Line(Point(5, 8), Point(7, 8)))
+           >>> p1.latus_rectum
+           Line2D(Point2D(0, 0), Point2D(1, 0))
+
```
Comment (Smit-create) [inline sympy/geometry/parabola.py:390]: It is failing doctest as you have not imported required functions/classes.
```suggestion

           >>> from sympy import Point, Line, Parabola
           >>> p1 = Parabola(Point(0, 0), Line(Point(5, 8), Point(7, 8)))
           >>> p1.latus_rectum
           Line2D(Point2D(0, 0), Point2D(1, 0))

```

**Comment by Smit-create** [comment]
Also, add the tests [here](https://github.com/sympy/sympy/blob/master/sympy/geometry/tests/test_polygon.py) to keep the coverage good.

**Comment by Mohitbalwani26** [comment]
@oscarbenjamin travis is showing "There was an error in the .travis.yml file from which we could not recover" so should I commit again?

**Comment by czgdp1807** [comment]
This is just a one liner code, IMHO, can be closed.

**Comment by Sc0rpi0n101** [comment]
Or we can merge this if this is done. 
I'm not sure what is left to be done here but the changes seem to be documented and tested properly. Although, the commits should be squashed before merging.

File: sympy/geometry/parabola.py [hunk 9]
```diff
@@ -376,6 +376,28 @@ def p_parameter(self):
...
+           ========
+
+           >>> from sympy import Point, Line, Parabola
+           >>> p1 = Parabola(Point(0, 0), Line(Point(5, 8), Point(7, 8)))
+           >>> p1.latus_rectum
+           Line2D(Point2D(0, 0), Point2D(1, 0))
+
+        """
+        s = self.directrix.slope
+        return Line(self.focus, slope=s)
```
Comment (smichr) [inline sympy/geometry/parabola.py:399]:
```suggestion
        return Line(self.focus, slope=self.directrix.slope)
```

File: sympy/geometry/tests/test_parabola.py [hunk 10]
```diff
@@ -71,6 +71,14 @@ def test_parabola_geom():
     assert pa10.focal_length == pa11.focal_length == sqrt((a - b) ** 2) / 2 # if a, b real == abs(a - b)/2
     assert pa11.vertex == Point(*pa10.vertex[::-1]) == Point(a,
                             a - sqrt((a - b)**2)*sign(a - b)/2) # change axis x->y, y->x on pa10
+    assert pa3.latus_rectum == Line(Point2D(0, 0), Point2D(0, 1))
+    assert pa4.latus_rectum == pa5.latus_rectum == Line(Point2D(3, 7), Point2D(4, 7))
+    assert pa6.latus_rectum == Line(Point2D(0, 4), Point2D(1, 4))
+    assert pa7.latus_rectum == Line(Point2D(3, 7), Point2D(3, 8))
+    assert pa8.latus_rectum == Line(Point2D(6, 0), Point2D(6, 1))
+    assert pa9.latus_rectum == Line(Point2D(6, 0), Point2D(6, 1))
```
Comment (smichr) [inline sympy/geometry/tests/test_parabola.py:79]:
```suggestion
```

Comment (smichr) [reply sympy/geometry/tests/test_parabola.py:79]: The two remaining tests pretty much cover all possibilities.

File: sympy/geometry/parabola.py [hunk 11]
```diff
@@ -376,6 +376,28 @@ def p_parameter(self):
             p = sign(self.focus.args[1] + y)
         return p * self.focal_length
 
+    @property
+    def latus_rectum(self):
+        """Returns the line passing through the focus
+           that is perpendicular to the axis of symmetry
+           Returns
```
Comment (smichr) [inline sympy/geometry/parabola.py:383]:
```suggestion

           Returns
```

File: sympy/geometry/parabola.py [hunk 12]
```diff
@@ -376,6 +376,28 @@ def p_parameter(self):
...
+           =======
+
+           latus_rectum : Line
+            Latus Rectum of Parabola
```
Comment (smichr) [inline sympy/geometry/parabola.py:387]:
```suggestion
               Latus Rectum of Parabola
```

File: sympy/geometry/parabola.py [hunk 13]
```diff
@@ -376,6 +376,28 @@ def p_parameter(self):
             p = sign(self.focus.args[1] + y)
         return p * self.focal_length
 
+    @property
+    def latus_rectum(self):
+        """Returns the line passing through the focus
```
Comment (smichr) [inline sympy/geometry/parabola.py:381]: Note that [Wolfram ](https://mathworld.wolfram.com/LatusRectum.html)gives a different definition and we should probably stick with that: 

> The latus rectum of a conic section is the chord through a focus parallel to the conic section directrix (Coxeter 1969). "Latus rectum" is a compound of the Latin latus, meaning "side," and rectum, meaning "straight."

File: sympy/geometry/parabola.py [hunk 14]
```diff
@@ -376,6 +376,29 @@ def p_parameter(self):
...
+           latus_rectum : Line
+               Latus Rectum of Parabola
+
+           Examples
+           ========
+
+           >>> from sympy import Point, Line, Parabola
+           >>> p1 = Parabola(Point(0, 0), Line(Point(5, 8), Point(7, 8)))
+           >>> p1.latus_rectum
+           Line2D(Point2D(0, 0), Point2D(1, 0))
```
Comment (smichr) [inline sympy/geometry/parabola.py:397]: This is not a chord, it's a line. It should return a Segment whose endpoints are where this line intersects the Parabola.

File: sympy/geometry/parabola.py [hunk 15]
```diff
@@ -376,6 +376,29 @@ def p_parameter(self):
...
+    def latus_rectum(self):
+        """The latus rectum of a conic section is the chord through
+           a focus parallel to the conic section directrix (Coxeter 1969).
+           "Latus rectum" is a compound of the Latin latus, meaning "side," and rectum, meaning "straight."
+
+           Returns
+           =======
+
+           latus_rectum : Line
+               Latus Rectum of Parabola
```
Comment (smichr) [inline sympy/geometry/parabola.py:389]:
```suggestion
        """return Segment parallel to directrix, passing through focus
```

**Comment by czgdp1807** [comment]
Closing this as of now. Please re-open it( or ask us to do so if you don't have the access) if you want to work on it.

**Pull request closed without merging**